From d609f8b6e1f556df55a1dfc14319db38d0a42418 Mon Sep 17 00:00:00 2001
From: Dmitri Epshtein <dima@marvell.com>
Date: Mon, 8 Jan 2018 16:46:01 +0200
Subject: [PATCH 2216/2241] dts: cp110: enable eip197 in SoC level only for
 70x0/80x0

In A8k+ we use EIP197d device in AP810 and
don't use EIP197b device in CP110
This patch prevent unneeded probing of EIP197b devices in cp110 for
A8K+ SoCs as 8082, 8084 and 8162

Change-Id: I3d0066bb87154c061d07ee471a301d79c1e9bc00
Signed-off-by: Dmitri Epshtein <dima@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48936
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/49446
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7020.dtsi  | 3 +++
 arch/arm64/boot/dts/marvell/armada-7040.dtsi  | 3 +++
 arch/arm64/boot/dts/marvell/armada-8020.dtsi  | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi  | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 2 +-
 5 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7020.dtsi b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
index a6ff9dd..55d4ee4 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
@@ -181,6 +181,9 @@
 			sar@400200 {
 				compatible = "marvell,a70x0-sample-at-reset";
 			};
+			cp0_eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 		pcie0@600000 {
 			msi-parent = <&gic_v2m0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index caef918..3a97767 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -191,6 +191,9 @@
 			sar@400200 {
 				compatible = "marvell,a70x0-sample-at-reset";
 			};
+			cp0_eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 		pcie0@600000 {
 			msi-parent = <&gic_v2m0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8020.dtsi b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
index cdb49a8..5bf51ae 100644
--- a/arch/arm64/boot/dts/marvell/armada-8020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
@@ -228,6 +228,9 @@
 			sar@400200 {
 				compatible = "marvell,a80x0-cp0-sample-at-reset";
 			};
+			cp0_eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
@@ -280,6 +283,9 @@
 			sar@400200 {
 				compatible = "marvell,a80x0-cp1-sample-at-reset";
 			};
+			cp1_eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 8e89d9d..72de71b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -241,6 +241,9 @@
 			sar@400200 {
 				compatible = "marvell,a80x0-cp0-sample-at-reset";
 			};
+			cp0_eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
@@ -293,6 +296,9 @@
 			sar@400200 {
 				compatible = "marvell,a80x0-cp1-sample-at-reset";
 			};
+			cp1_eip197: eip197@800000 {
+				status = "okay";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 154b060..0bca4c6 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -542,7 +542,7 @@
 						"ring2", "ring3", "eip_out";
 				clocks = <&CP110_LABEL(syscon0) 1 26>;
 				cell-index = <CP110_EIP197_INDEX>;
-				status = "okay";
+				status = "disabled";
 			};
 
 			CP110_LABEL(comphy): comphy {
-- 
2.7.4

